mos_tests/rabbitmq_oslo/test_restarts.py

```diff
@@ -0,0 +1,183 @@
+    :param remote: SSH connection point to controller.
+    """
+    config_vars = {
+        'repo': settings.RABBITOSLO_REPO,
+        'pkg': settings.RABBITOSLO_PKG,
+        'root_path': '/root/',
+        'nova_user': 'nova'}
+    # like: /root/oslo.messaging-check-tool/
+    config_vars['repo_path'] = '{0}{1}/'.format(
+        config_vars['root_path'], config_vars['repo'].split('/')[-1][:-4])
+    # like: /root/oslo.messaging-check-tool/oslo_msg_check.conf
+    config_vars['conf_file_path'] = '{}oslo_msg_check.conf'.format(
+        config_vars['repo_path'])
+    # get password of nova user (the same on all controllers)
+    cmd = "grep '^rabbit_password' /etc/nova/nova.conf | awk '{print $3}'"
+    config_vars['nova_pass'] = remote.check_call(cmd)['stdout'][0].strip()
+    return config_vars
+
+
+def install_oslomessagingchecktool(remote, **kwargs):
+    """Install 'oslo.messaging-check-tool' on controller.
+    https://github.com/dmitrymex/oslo.messaging-check-tool
+    :param remote: SSH connection point to controller
+    """
+    cmd = ("apt-get update && "
+           "apt-get install git python-pip -y && "
+           "cd {root_path} && "
+           "git clone {repo} && "
+           "cd {repo_path} && "
+           "pip install -r requirements.txt -r test-requirements.txt && "
+           "dpkg -i {pkg} || "
+           "apt-get -f install -y".format(**kwargs))
+    logger.debug('Install "oslo.messaging-check-tool" on controller')
+    return remote.check_call(cmd)
+
+
+def configure_oslomessagingchecktool(remote, ctrl_ips, nova_user, nova_pass,
+                                     conf_file_path):
+    """Write configuration file on controller
+    :param remote: SSH connection point to controller;
+    :param ctrl_ips: List of controllers IPs;
+    :param nova_user: Name of Rabbit admin;
+    :param nova_pass: Password of Rabbit admin;
+    :param conf_file_path: Path where config file will be written.
+    """
+    # Create config file for "oslo.messaging-check-tool"
+    configs = (
+        "[DEFAULT]\n"
+        "debug=true\n"
+        "[oslo_messaging_rabbit]\n"
+        "rabbit_hosts = {rabbit_hosts}\n"
+        "rabbit_userid = {nova_user}\n"
+        "rabbit_password = {nova_pass}\n".format(
+            rabbit_hosts=', '.join([(x + ':5673') for x in ctrl_ips]),
+            nova_user=nova_user,
+            nova_pass=nova_pass))
+    # Write config to file on controller
+    logger.debug('Write "oslo.messaging-check-tool" config file to controller')
+    with remote.open(conf_file_path, 'w') as f:
+        f.write(configs)
+
+
+def restart_rabbitmq_serv(env, remote=None):
+    """Restart rabbitmq-server service on one or all controllers
+    :param env: Environment
+    :param remote: SSH connection point to controller.
+        Leave empty if you want to restart service on all controllers.
+    """
+    # 'sleep' is to wait for service startup
+    restart_cmd = 'service rabbitmq-server restart && sleep 20'
```

Comment:
I think the 'sleep' here is not necessary, since 'service rabbitmq-server restart ' returns exit code after a complete restart. otherwise it is necessary to make a more complex checking using a status checking loop.